### PR TITLE
Merge #1068 to cats

### PR DESCRIPTION
**Pull request by rossabaker**
This one got a little chewier than most merges, so I'm putting it through review.

### CODE REVIEW

File: core/src/main/scala/org/http4s/util/StreamApp.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      .append(eval_(halted.set(true)))
+
+    sys.addShutdownHook {
+      requestShutdown.unsafeRunAsync(_ => ())
+      halted.discrete.takeWhile(_ == false).run.unsafeRun
+    }
+
+    p.run.attempt.unsafeRun match {
+      case Left(t) =>
+        t.printStackTrace()
```
Comment (bfritz) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: Was `printStackTrace()`, instead of using a logger, intentional here (and in #1068)?

Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: This class might be useful to fs2, which I don't think has a logger.   The specialization on Task might make it less attractive there.  Until it gets offered up there, there's no good reason not to use a logger here.

Comment (bfritz) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: So a case where fs2 and http4s-core are in the classpath, no slf4j binding is in classpath, and logging calls become no-op?

Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: Is the default no-op or stdout?

Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: You get that nasty warning from slf4j.  Ignore at your own peril, I say.

Comment (bfritz) [inline core/src/main/scala/org/http4s/util/StreamApp.scala]: Agreed and no-op [starting with](https://www.slf4j.org/manual.html) slf4j 1.6.0.

**Pull request merged**